lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/LibertyDiagnosticParticipant.java

```diff
@@ -175,11 +182,28 @@ private void validateIncludeLocation(DOMDocument domDocument, List<Diagnostic> d
                 }
                 diagnosticsList.add(new Diagnostic(range, MISSING_FILE_MESSAGE, DiagnosticSeverity.Warning, LIBERTY_LEMMINX_SOURCE, MISSING_FILE_CODE));
             }
+            validateFileOrDirIncludeLocation(configFile, isLibertyDirectory, range, diagnosticsList);
         } catch (IllegalArgumentException e) {
             diagnosticsList.add(new Diagnostic(range, MISSING_FILE_MESSAGE, DiagnosticSeverity.Warning, "liberty-lemminx-exception", MISSING_FILE_CODE));
         }
     }
 
+    /**
+     * Checks if specified file or dir is the correct filetype.
+     * Adds diagnostics if it is mismatched.
+     * @param f - <include> location file
+     * @param isLibertyDirectory - whether Liberty considers this specified as a dir (path ends in slash)
+     * @param range - Range to apply diagnostic message
+     * @param diagnosticsList
+     */
+    private void validateFileOrDirIncludeLocation(File f, boolean isLibertyDirectory, Range range, List<Diagnostic> diagnosticsList) {
+        if (f.isFile() && isLibertyDirectory) {
```

Comment:
Does resolving a file with an ending slash work when it is a file? So `new File("myconfgdir/server.xml/")` returns a valid `File` object?

Reply:
It does. The tests also include this scenario